### ISSUE TITLE
Support `--auth-proxy` for `rclone serve http`

### ISSUE THREAD

**Issue by baurmatt** [rclone/rclone#6779]
#### The associated forum post URL from `https://forum.rclone.org`

https://forum.rclone.org/t/make-backend-configuration-configurable-per-http-request-on-serve-http/36202/3

#### What is your current rclone version (output from `rclone version`)?

```bash
rclone v1.61.1
- os/version: darwin 13.0 (64 bit)
- os/kernel: 22.1.0 (arm64)
- os/type: darwin
- os/arch: arm64
- go/version: go1.19.4
- go/linking: dynamic
- go/tags: none
```

#### What problem are you are trying to solve?

Dynamically configure [--s3-version-at](https://github.com/rclone/rclone/commit/4344a3e2ea8e78127371a5feb476344171898950) on a per request basis

#### How do you think rclone should be changed to solve that?

As suggested by @ncw in the forum: Support [`--auth-proxy`](https://rclone.org/commands/rclone_serve_webdav/#auth-proxy) for `rclone serve http`.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by baurmatt** [comment]
I think I've got a working implementation and might provide a PR later.

**Comment by ncw** [comment]
Fantastic! A PR would be great.

**Comment by baurmatt** [comment]
Actually, this has already been implemented and merged with https://github.com/rclone/rclone/pull/6780

Released with [1.62.0](https://github.com/rclone/rclone/releases/tag/v1.62.0)
